### PR TITLE
fix deprecations

### PR DESCRIPTION
**Pull request by abhijithch**
Fixes deprecations on `Julia v0.5`.

### CODE REVIEW

File: src/zlib_h.jl [hunk 1]
```diff
@@ -1,11 +1,14 @@
 # general zlib constants, definitions
 
-@unix_only    const _zlib = "libz"
-@windows_only const _zlib = "zlib1"
+@static if is_unix()
+    const _zlib = "libz"
+elseif iswindows()
```
Comment (kmsquire) [inline src/zlib_h.jl]: This currently only exists on Julia v0.4 and needs to be added to Compat.jl.

Cc: @vtjnash

Comment (quinnj) [inline src/zlib_h.jl]: It's in Compat: https://github.com/JuliaLang/Compat.jl/blob/546eccbeac2ffe3f80ac4854b7d02e853ead2565/src/Compat.jl#L1167

Comment (kmsquire) [inline src/zlib_h.jl]: Ah, it's misspelled here.  Should be `is_windows`.

File: src/zlib_h.jl [hunk 2]
```diff
@@ -1,11 +1,14 @@
 # general zlib constants, definitions
 
-@unix_only    const _zlib = "libz"
-@windows_only const _zlib = "zlib1"
+@static if is_unix()
```
Comment (vtjnash) [inline src/zlib_h.jl]: `@static` is redundant

Comment (tkelman) [inline src/zlib_h.jl]: and underdocumented

Comment (abhijithch) [inline src/zlib_h.jl]: @vtjnash, the deprecation message suggests to use `@static` (`WARNING: `@windows_only` is deprecated, use @static if is_windows() instead`). Maybe we should update that as well. 

@tkelman, could you please clarify what is underdocumented?

**Comment by abhijithch** [comment]
@tkelman, is this good to merge?

**Comment by kmsquire** [comment]
I'll merge later today if there are no additional comments. I believe the appveyor failure is just because we haven't spent enough time getting appveyor working correctly.

**Comment by abhijithch** [comment]
OK! Thanks @kmsquire.

**Pull request merged**